File: game/MMO/NFServerLogic/NFProxyServer/NFProxyClientPlugin/NFProxySession.cpp
// -------------------------------------------------------------------------
//    @FileName         :    NFProxySession.cpp
//    @Author           :    gaoyi
//    @Date             :    2022/9/29
//    @Email			:    445267987@qq.com
//    @Module           :    NFProxySession
//
// -------------------------------------------------------------------------

#include "NFProxySession.h"

uint64_t NFProxySession::GetLinkId() const
{
    return m_linkId;
}

void NFProxySession::SetLinkId(uint64_t linkId)
{
    m_linkId = linkId;
}

uint64_t NFProxySession::GetUid() const
{
    return m_uid;
}

void NFProxySession::SetUid(uint64_t playerId)
{
    NFProxySession::m_uid = playerId;
}

const string &NFProxySession::GetIpAddr() const
{
    return m_ipAddr;
}

void NFProxySession::SetIpAddr(const string &ipAddr)
{
    m_ipAddr = ipAddr;
}

uint32_t NFProxySession::GetPort() const
{
    return m_port;
}

void NFProxySession::SetPort(uint32_t port)
{
    m_port = port;
}

uint64_t NFProxySession::GetLastHeartBeatTime() const
{
    return m_lastHeartBeatTime;
}

void NFProxySession::SetLastHeartBeatTime(uint64_t lastHeartBeatTime)
{
    m_lastHeartBeatTime = lastHeartBeatTime;
}

uint64_t NFProxySession::GetLastActiveTime() const
{
    return m_lastActiveTime;
}

void NFProxySession::SetLastActiveTime(uint64_t lastActiveTime)
{
    m_lastActiveTime = lastActiveTime;
}

uint32_t NFProxySession::GetTimeOut() const
{
    return m_timeOut;
}

void NFProxySession::SetTimeOut(uint32_t timeOut)
{
    m_timeOut = timeOut;
}

int NFProxySession::CheckPkgRate(NFPackageConfig *pConfig, int iMsgID, int &count, int &interval)
{
    return m_packageMng.CheckPkgRate(pConfig, iMsgID, count, interval);
}

int NFProxySession::AddPkgStatistic(int iMsgID, uint64_t roleID, uint64_t linkId)
{
    return m_packageMng.AddPkgStatistic(iMsgID, roleID, linkId);
}

File: Install/ScriptModule/Fish/ProxyServer/NFProxySession.lua

LuaNFrame.Class("NFProxySession")

NFProxySession.mapData = NFMap.New()

function NFProxySession.Create(linkId)
    local session = NFProxySession.New()
    NFProxySession.mapData:Insert(linkId, session)
    return session
end

function NFProxySession.GetSession(linkId)
    return NFProxySession.mapData:Find(linkId)
end

function NFProxySession.DeleteSession(linkId)
    NFProxySession.mapData:Remove(linkId)
end

function NFProxySession:Ctor()
    self.m_linkId = 0;
    self.m_playerId = 0;
    self.m_lastHeartBeatTime = 0;
    self.m_port = 0;
    self.m_timeOut = 0;
    self.m_lastActiveTime = 0;
    self.m_ipAddr = "127.0.0.1";
end

function NFProxySession:GetLinkId()
    return self.m_linkId
end

 function NFProxySession:SetLinkId(mLinkId)
    self.m_linkId = mLinkId
 end

 function NFProxySession:GetPlayerId()
    return self.m_playerId
 end

 function NFProxySession:SetPlayerId(playerId)
    self.m_playerId = playerId
 end

 function NFProxySession:GetIpAddr()
    return self.m_ipAddr
 end

 function NFProxySession: SetIpAddr(mIpAddr)
    self.m_ipAddr = mIpAddr
 end

 function NFProxySession:GetPort()
    return self.m_port 
 end

 function NFProxySession:SetPort(mPort)
    self.m_port  = mPort
 end

 function NFProxySession:GetLastHeartBeatTime()
    return self.m_lastHeartBeatTime
 end

function NFProxySession:SetLastHeartBeatTime(mLastHeartBeatTime)
    self.m_lastHeartBeatTime = mLastHeartBeatTime
end

function NFProxySession:GetLastActiveTime()
    return self.m_lastActiveTime
end

function NFProxySession:SetLastActiveTime(mLastActiveTime)
    self.m_lastActiveTime = mLastActiveTime
end

function NFProxySession:GetTimeOut()
    return self.m_timeOut
end

function NFProxySession:SetTimeOut(mTimeOut)
    self.m_timeOut = mTimeOut
end